### ISSUE TITLE
Live tailing was stopped due to following error: undefined

### ISSUE THREAD

**Issue by ldaneliukas** [grafana/loki#7153]
An `undefined` error is **sometimes** displayed when attempting to use live tailing of logs via Grafana Explore UI with a datasource that spans multiple tenants. Seemingly the same time, Loki Read logs the following (not completely sure if this is related):
```
level=warn ts=2022-09-14T10:41:18.304139412Z caller=http.go:273 msg="error getting tenant id" err="multiple org IDs present"
```

The query in question is for the logs of 3 containers (all of which belong to the same tenant): `{container=~"thanos-compact-d1-18mo|thanos-compact-d1|thanos-compact-d1-6mo"} |= `` `

Live tailing the above query via a datasource that contains a single tenant always works.
Live tailing the above query via a datasource that contains multiple tenants **sometimes** works and **sometimes** throws the above error in Grafana.

**To Reproduce**
1. Started Loki v2.6.1
2. Started Grafana v9.1.4
3. Add Loki datasource with multiple tenants separated by `|` via the X-Scope-OrgID header
4. Query logs from multiple sources (same tenant) via explore UI.
5. Turn on live tailing

**Expected behavior**
All logs matching the query to be live tailed

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: Helm

**Screenshots, Promtail config, or terminal output**
<img width="921" alt="image" src="https://user-images.githubusercontent.com/10710427/190134967-65bd13f9-e279-46bb-8326-013970d794fb.png">

**Comment by i5okie** [comment]
Did you get this resolved?
I just started to run into this issue.
Didn't seem to have this issue until I configured S3 storage though.

Using loki simple scalable helm chart.
my initial suspicion is that this has something to do with local caching not being configured properly. But really not sure yet.

**Comment by LDaneliukas** [comment]
No, we haven't. We were using S3 as the object-storage for our Loki since launch and initially, we were running on pretty much defaults apart from the object-storage settings - that's when we saw the issue appear.

We're also using simple-scalable, or [Loki](https://github.com/grafana/loki/tree/main/production/helm/loki) v3.2.1 to be exact.

**Comment by verejoel** [comment]
We’re also seeing this issue. Live tailing seems to be broken when using a data source with multiple tenants in the header.

**Comment by kawhicurry** [comment]
Same issue with simple-scalable Loki. But only one tenants.

**Comment by lhbarry** [comment]
Same, simple-scalable, minIO with single tenant

**Comment by boagg** [comment]
Same problem on my side, i no longer have the error by configuring the following headers at the level of the LOKI datasource:

```
Connection: Upgrade
Upgrade: websocket
```

See: #https://github.com/grafana/grafana/issues/22905#issuecomment-628442315

**Comment by kfkawalec** [comment]
In my case the problem was missing "X-Scope-OrgID" header. Normal logs worked without this header, but not live!

**Comment by oleksii-boiko-ua** [comment]
> We’re also seeing this issue. Live tailing seems to be broken when using a data source with multiple tenants in the header.

we are having exact same issues when header contains more than 1 tenant id

**Comment by kacey-lunacare** [comment]
>> We’re also seeing this issue. Live tailing seems to be broken when using a data source with multiple tenants in the header.

>we are having exact same issues when X-Scope-OrgID header contains more than 1 tenant id.
>Grafana v9.5.3
>Loki v2.8.2

That's by design :( 
https://grafana.com/docs/loki/latest/operations/multi-tenancy/#multi-tenant-queries
>Only query endpoints support multi-tenant calls. Calls to GET /loki/api/v1/tail and POST /loki/api/v1/push will return an HTTP 400 error if more than one tenant is defined in the HTTP header.

**Comment by oleksii-boiko-ua** [comment]
> Any solution how to use live log in grafana when datasource with X-Scope-OrgID contains more than 1 tenant id?

best workaround i found so far is to use separate datasource per tenant

**Comment by pdostal** [comment]
I've reproduced this in the current version against single updated Loki with a simple query:
![image](https://github.com/grafana/loki/assets/1254493/17f982b5-be2a-406c-a595-6a452737a2b5)

**Comment by Jeremy-Boyle** [comment]
https://github.com/grafana/grafana/issues/22905#issuecomment-1644120475

**Comment by beatstream69** [comment]
Same error, but another source:
I have Grafana 10.2.1 and Loki 2.9.2 behind nginx proxy with basic auth.
Querying logs from Grafana works normal, but Live tailing throws an error `Live tailing was stopped due to following error: undefined` and in browser logs I see `401 Unauthorized` when connecting to wss.
X-Scope-OrgID header has been set in datasource settings - without it not working at all.
Trying to set Connection and Upgrade headers without success.
Setting Host, Connection and X-Scope-OrgID explicitly in nginx config not helps.

PS: Finally I found the issue in nginx config - my location directive was `location ~ /loki/api/.*` which was intercepting query to grafana `/grafana/api/datasources/proxy/uid/.../loki/api/v1/tail`.
Final nginx config:
```
location = /loki/api/v1/push {
        auth_basic "Restricted area";
        auth_basic_user_file /etc/nginx/.htpasswd;
        proxy_pass $loki_write;
}
location ~ ^/loki/api/(.*) {
        auth_basic "Restricted area";
        auth_basic_user_file /etc/nginx/.htpasswd;
        proxy_pass $loki_read;
}
```

**Comment by GirishMS187** [comment]
Live logs are coming, but they are not consistent. After a while, the logs stop due to the following issue.
```
Query error
Live tailing was stopped due to following error: undefined```

**Comment by Shashwatpathak05** [comment]
same issue

**Comment by algo7** [comment]
> I was able to fix this via,
> 
> Also have muti tenancy.
> 
> [grafana/grafana#22905 (comment)](https://github.com/grafana/grafana/issues/22905#issuecomment-1644120475)

Can you shine some light one this? We are having the exact issue with multiple tenant ID. But according to this issue here: https://github.com/grafana/grafana/issues/69465

Live tailing for multi-tenancy is not supported

**Comment by VenkataSalopantula** [comment]
same issue

**Comment by velykov** [comment]
same issue

**Comment by karthick-encipherhealth** [comment]
I'm also getting the same error. I'm using Azure Managed Grafana, and I have installed Loki via Helm. Loki (distributed) is connected to an Azure Storage Account.
`apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: logs-api
  namespace: observability
  annotations:
    # WebSocket support
    nginx.ingress.kubernetes.io/proxy-http-version: "1.1"
    nginx.ingress.kubernetes.io/enable-websocket: "true"
    nginx.ingress.kubernetes.io/configuration-snippet: |
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
    cert-manager.io/cluster-issuer: xxx-yyy
    nginx.ingress.kubernetes.io/use-regex: "true"
    nginx.ingress.kubernetes.io/enable-cors: "true"
    nginx.ingress.kubernetes.io/cors-allow-origin: "*"
    nginx.ingress.kubernetes.io/cors-allow-headers: "*"
    nginx.ingress.kubernetes.io/proxy-read-timeout: "600"
    nginx.ingress.kubernetes.io/proxy-send-timeout: "600"
    nginx.ingress.kubernetes.io/proxy-connect-timeout: "600"
    nginx.ingress.kubernetes.io/proxy-body-size: "100m"

spec:
  ingressClassName: nginx
  tls:
    - hosts:
        - xx.yyyy.in
      secretName: logs-api
  rules:
    - host: xx.yyyy.in
      http:
        paths:
          - path: /loki/
            pathType: ImplementationSpecific
            backend:
              service:
                name: loki-gateway
                port:
                  number: 80`

this is my ingress and this is my values yaml of loki
`queryFrontend:
  replicas: 8
  maxUnavailable: 2
  max_concurrent_tail_requests: 50
  frontend_worker:
    frontend_address: loki-query-frontend:9095

gateway:
  enabled: true
  ingress:
    enabled: true
    hosts:
      - host: xxx.yyyy.in
        paths:
          - path: /loki/
            pathType: Prefix
  proxy:
    extraConfigs: |
      location ~ ^/loki/(.*) {
        proxy_pass http://loki-query-frontend:3100/$1;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_read_timeout 3600s;
        proxy_redirect off;
      }`


if anyone used this method, suggest me a solution
